### ISSUE TITLE
Python: Filters for FunctionCallBehavior.EnableFunctions should be optional

### ISSUE THREAD

**Issue by matthewbolanos** [microsoft/semantic-kernel#6924]
Today, the filters property is required, even if its empty (see below)

```python
execution_settings.function_call_behavior = FunctionCallBehavior.EnableFunctions(auto_invoke=True, filters={})
```

I should be able to just write...

```python
execution_settings.function_call_behavior = FunctionCallBehavior.EnableFunctions(auto_invoke=True)
```

**Comment by moonbox3** [comment]
We're moving to support `FunctionChoiceBehavior` (#6910) where the filters for each class method are optional. However, for existing functionality, and for those users who haven't migrated from `FunctionCallBehavior` -> `FunctionChoiceBehavior` we can make the update to make the default filters, if not passed, as an empty dict.

**Comment by moonbox3** [comment]
Closed as of #6910.
